### ISSUE TITLE
Validation script

### ISSUE THREAD

**Issue by andrewdalpino** [andrewdalpino/esm2-function-classifier#2]
Let's create a script that takes the pretrained models from HuggingFace Hub (ex. https://huggingface.co/andrewdalpino/ESM2-35M-Protein-Cellular-Component) and tests them on a hold-out dataset. The idea is to get a deeper understanding of how well the model performs and generalizes to unseen data. We can get as basic or fancy as you'd like. This will be the "final" validation step before the model is put into production and takes precedence over any validation done during training to monitor overfitting.

Rough Approach:

1. Load pretrained model and tokenizer
2. Load the test split of the AmiGO dataset
3. Predict the GO terms of the test samples
4. Compare predictions to ground-truth
5. ???
6. Profit!

Questions/Concerns?

- Is it ok that the test set includes similar sequences but from a different species?
- Should we apply weighting for terms the deeper they are in the GO hierarchy?

**Comment by andrewdalpino** [comment]
I think it would be cool to be able to visualize the predicted vs. ground-truth subgraphs too.
